CMakeLists.txt

```diff
@@ -5,46 +5,69 @@ project(match-theory VERSION 0.1.0)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-add_compile_options(-Wall -Wextra -pedantic)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Does this also match against Clang?

Reply:
It does not. Does clang supports the same flags? I found that MSVC does not